src/chainstate/stacks/db/mod.rs

```diff
@@ -609,11 +609,20 @@ impl StacksChainState {
         {
             let mut clarity_tx = chainstate.block_begin(&NULL_BURN_STATE_DB, &BURNCHAIN_BOOT_CONSENSUS_HASH, &BOOT_BLOCK_HASH, &FIRST_BURNCHAIN_CONSENSUS_HASH, &FIRST_STACKS_BLOCK_HASH);
             for (boot_code_name, boot_code_contract) in STACKS_BOOT_CODE.iter() {
+                let boot_code_contract_string = 
+                    if *boot_code_name == "pox" {
```

Comment:
Instead of having magic constants embedded within the code, can we just instead create `STACKS_BOOT_CODE_MAINNET` and `STACKS_BOOT_CODE_TESTNET` arrays, and pick which one to use based on the value of `mainnet`?

Reply:
Sure

Reply:
Actually, doing that will result in some code magnification (replicating the `.clar` file contents), is that preferable to checking the boot contract?

Reply:
The fundamental problem here is that a Clarity contract can't read constants in another Clarity contract.  If this were the case, then we could deploy `pox-constants.clar` as its own contract, and have `pox.clar` read from it.  The absence of this feature forces us to concatenate the two contracts at runtime and deploy them as a single contract.

The thing I'm most concerned with is avoiding peppering the codebase with special-case code paths for the `pox` contract.  It's annoying enough that we have to do this for the `handle_special_cases` codepath in the VM for locking/unlocking, but I don't know what else to do about that.  Is there any way we can avoid creating more special-case code paths by structuring our data a bit more intelligently?

Reply:
I'll just use a lazy_static here.

Reply:
Addressed in 9d5d1e0d3